### PR TITLE
Ext proc optimization

### PR DESCRIPTION
**Pull request by DiazAlan**
Commit Message: Change API call on ext_proc to more efficient new implementation
Additional Description: Following the PR #[29199](https://github.com/envoyproxy/envoy/pull/29199), we are replacing the API call in ext_proc so it's not hashing the whole proto on each request.

### CODE REVIEW

**Comment by DiazAlan** [comment]
@ggreenway This is another PR to continue moving over filters to the new getOrCreateRawAsyncClient method.

**Review by ggreenway — CHANGES_REQUESTED** [review]
A few nits, but looks good overall

/wait

File: envoy/grpc/async_client_manager.h [hunk 1]
```diff
@@ -34,6 +34,8 @@ using AsyncClientFactoryPtr = std::unique_ptr<AsyncClientFactory>;
 
 class GrpcServiceConfigWithHashKey {
 public:
+  GrpcServiceConfigWithHashKey(){};
```
[the remark this answers is not in the document]
Comment (DiazAlan) [reply envoy/grpc/async_client_manager.h]: Done

File: envoy/grpc/async_client_manager.h [hunk 2]
```diff
@@ -53,9 +55,14 @@ class GrpcServiceConfigWithHashKey {
 
   const envoy::config::core::v3::GrpcService& config() const { return config_; }
 
+  void mergeConfig(const envoy::config::core::v3::GrpcService g) {
```
[the remark this answers is not in the document]
Comment (DiazAlan) [reply envoy/grpc/async_client_manager.h]: I changed it to setConfig because I feel like that fits the use case better than update.

**Review by ggreenway — APPROVED** [review]

**Pull request merged**